_posts/2016-04-10-running-multiple-apps-in-umbrella-project.md

```diff
@@ -0,0 +1,114 @@
+Off-the-shelf Opensource E-commerce application for building an online store.
+>
+Provides an Extension Framework to support features not included in core as extensions.
+>
+Strives for unobtrusive parallel development of NectarCommerce and Extensions
+
+NectarCommerce is committed to providing a ready-to-use e-commerce solution but the definition of 100% is different under different business domains. It aims to solve common use-cases as part of the project and relying on extension framework to tap the rest.
+
+# Elixir Umbrella Project
+
+[Elixir Umbrella Project](http://elixir-lang.org/getting-started/mix-otp/dependencies-and-umbrella-apps.html#umbrella-projects
+) is a convenience to help you organize and manage your applications. Applications inside the apps directory are still decoupled from each other. Dependencies between them must be explicitly listed. This allows them to be developed together, but compiled, tested and deployed independently if desired.
+
+As NectarCommerce extensions are envisioned to use with Nectar and would be dependent on Nectar directly or indirectly, we chose umbrella project to have Nectar, extensions\_manager, user\_app, and extensions apps as part of it.
+
+We plan to use `phoenix app` for extensions where web-component is needed and which would be part of umbrella project.
+
+The `user_app` would be a resting ground where Nectar and extensions are evaluated for compile time requirements and prepared to be used together.
+
+As of now, `user_app` has following responsibilities:
+
+- [defining nectar as umbrella dependency](https://github.com/vinsol/nectarcommerce/pull/47/files#diff-cfadcd558b7d7d0fa0bb1b43ff49ba98R44)
+- [defining available extensions as umbrella dependencies](https://github.com/vinsol/nectarcommerce/pull/47/files#diff-cfadcd558b7d7d0fa0bb1b43ff49ba98R45)
+- [acting as a pass-through for all the outbound requests](https://github.com/vinsol/nectarcommerce/pull/47/files#diff-02eb191c0680951b40ba159f4163a7c6R21)
+
+## Running Multiple Phoenix Applications together in an umbrella ##
+
+Phoenix applications are in the end just another OTP application, with a few modifications and configuration tweaks we can expect them to run together in same umbrella project. We need to make sure that
+
+1. They share the same configuration for database, while not mandatory if all the applications are utilizing the same databases, keeping the configuration in one place is a good practice. We can create a ```shared_config.exs``` at root of umbrella with some code like this:
+	
+	
+	```elixir
+	config :shared, :db_config,
+	  username: "shared_username",
+	  pool_size: 200
+	```
+	
+	And, then in each individual application at the end of file:
+	
+	```elixir
+	shared_db_config = Mix.Config.read!("../../shared_config.exs")[:shared][:db_config]
+	config :app, App.Repo, shared_db_config
+	```
+	
+	Obviously this is use case dependent some applications may not share a database and only rely on each other for method calls.
```

Comment:
Obviously this is use case dependent, some applications may not share a database and only rely on each other for method calls.